### DISCUSSION TITLE
Debugging 502 errors is not possible, how can I deploy to production with confidence?

### DISCUSSION THREAD

**Discussion by shaneosullivan** [vercel/vercel#5772] [Help]
I'm seeing 502 errors from my service where rendering a simple page is timing out after 60s, even though it usually just takes a couple of seconds. It doesn't reproduce locally, and no logs show up in the console.  This appears to be the extent of the debuggability on Vercel - the docs at https://vercel.com/knowledge/how-to-debug-a-502-error is oddly content free given it's discussing a fundamental breakage of a service.

How do others deploy a production service with confidence?  Do you have ways of tracking memory & CPU usage, outgoing DB connections with DB Pool statistics?  Should I give up on using Vercel for production and just go back to the Hobby plan for development?

**Comment by HassanAlmus** [comment]
same issue here
